staging/src/k8s.io/legacy-cloud-providers/gce/gce.go

```diff
@@ -672,11 +698,13 @@ func (g *Cloud) NetworkURL() string {
 
 // SubnetworkURL returns the subnetwork url
 func (g *Cloud) SubnetworkURL() string {
+	g.subnetworkURLAndIsLegacyNetworkInitializer.Do(g.initializeSubnetworkURLAndIsLegacyNetwork)
 	return g.subnetworkURL
 }
 
 // IsLegacyNetwork returns true if the cluster is still running a legacy network configuration.
 func (g *Cloud) IsLegacyNetwork() bool {
```

Comment:
https://github.com/kubernetes/kubernetes/pull/90218/files#diff-bdd9c58b7687356fb858912c3ae53cbcR78 (gce_loadbalancer_internal.go ine 76) is still using g.isLegacyNetwork directly, potentially without initialization.

Preferably, we should migrate the potentially uninitialized fields to separete struct to hide them, but this will be larger change that potentially won't be cherrypickable.

If above is not doable, I think it makes sense to at least rename g.subnetworkURL to g.unsafeSubnetworkURL and g.isLegacyNetwork to g.unsafeLegacyNetwork to detect all places that can use those fields directly and make it harder to use that field incorrectly in future.